src/core/task/tools/handlers/WriteToFileToolHandler.ts

```diff
@@ -445,7 +285,142 @@ export class WriteToFileToolHandler implements IFullyManagedTool {
 			// Reset diff view on error
```

Comment:
@saoudrizwan Right above this code on line 283 there should be a reset for diffview edits and we should keep it here too to avoid weird overwrite issues etc. 

The [original code has it](https://github.com/cline/cline/blob/bd526cb2d2244efab79bb16942439d5edb38c782/src/core/task/ToolExecutor.ts#L634C35-L634C40)

Reply:
We do call `await config.services.diffViewProvider.reset()` on line 261 before returning the response.